pyrealm/demography/tmodel.py

```diff
@@ -445,6 +445,67 @@ def calculate_foliar_respiration(
     return _enforce_2D(whole_crown_gpp * resp_f)
 
 
+def calculate_gpp_topslice(
+    gpp_topslice: NDArray[np.float64],
+    whole_crown_gpp: NDArray[np.float64],
+    validate: bool = True,
+) -> NDArray[np.float64]:
+    r"""Calculate gpp topslice.
+
+    Calculates a fixed proportion of the total GPP for the crown that is removed before
+    further GPP allocation. This is intended as a helper variable for T Model users to
+    simulate processes not included in the T Model such as root exudation or active
+    nutrient servicing for mycorriza fungi.
+
+    NOTE: This is a naieve calculation method that is not part of the T model. If values
+    for GPP topslice are zero it will have no impact on the T Model calculations.
```

Comment:
Here and below, we can use this RST syntax to style the note as an admonition, which just helps draw attention to it. See e.g. https://pyrealm.readthedocs.io/en/v2.0.0-rc.1/api/pmodel_api.html#pyrealm.pmodel.optimal_chi.OptimalChiLavergne20C4


```suggestion
    .. NOTE:: 
        
        This is a naive calculation method that is not part of the T model. If values
        for GPP topslice are zero it will have no impact on the T Model calculations.
```